core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SecurityScanGithubWorkflowProbe.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 Jenkins Infra
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(SecurityScanGithubWorkflowProbe.ORDER)
+public class SecurityScanGithubWorkflowProbe extends Probe {
+    public static final int ORDER = LastCommitDateProbe.ORDER + 100;
```

Comment:
This is coming from Spring Framework. See https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/annotation/Order.html

Reply:
Yeah, for the annotation I got it, I was rather wondering about how we are defining the values in there? Is there some kind of meaning/strategy in the way you define the `ORDER` values in variables?

Reply:
ok so no it's not documented. 
The idea was rather than having increment of 1, I choose to go with increment of 100 so that when listing the probes, the order is more obvious.
The increment of 10 are reserved to probes that have a strict link with its predecessor.

The idea is also to say "probe B needs probe A, so increment probe A by 100 for probe B". If probe C also needs probe A, do the same. If probe C also need probe B then let's use probe B as the base.

This may make only sense in my head..

Reply:
Makes sense, but it would totally deserve a little bit of documentation =)

If it was me, I would be tempted to create a small utility class for ORDER, add all your explanation as Javadoc, and add some constants to use such as `REQUIRES_PREDECESSOR = 10` or `NEXT_PROBE = 100` for example, and use that here for better readability, like `ORDER = LastCommitDateProbe.ORDER + NEXT_PROBE;`.

WDYT?